components/carousel/module/CarouselAgnostic.vue

```diff
@@ -95,6 +122,32 @@ const totalDots = computed(() => {
 const dotHelper = computed(() =>
   slider.value && totalDots.value > 0 ? [...Array(totalDots.value).keys()] : []
 )
+
+const slideIndex = ref(3)
+
+const sliderSettings = computed(() => {
+  if (slider.value) {
+    const { track, options, slides } = slider.value
+    const abs = track.details.abs
+    const perView = options.slides.perView
+    const leftArrowValid = abs !== 0
+    const rightArrowValid = abs + perView < slides.length
+    const leftCarouselIndex = Math.max(abs - slideIndex.value, 0)
+    const rightCarouselIndex = Math.min(
+      abs + slideIndex.value,
+      slides.length - perView
+    )
+
+    return {
+      leftArrowValid,
+      rightArrowValid,
+      leftCarouselIndex,
+      rightCarouselIndex,
+    }
+  } else {
+    return {}
+  }
```

Comment:
do we need this empty return?

Reply:
yes, otherwise it's sometimes undefined (slider check is needed, because it loads slowly)